packages/itwin/measure-tools/src/api/SheetMeasurementHelper.ts

```diff
@@ -159,16 +159,50 @@ export namespace SheetMeasurementsHelper {
    * @param allowedDrawingTypes
    * @returns
    */
-  export function checkIfAllowedDrawingType(ev: BeButtonEvent, allowedDrawingTypes: DrawingType[]) {
-    if (!ev.viewport)
+  export function checkIfAllowedDrawingType(viewport: ScreenViewport | undefined, point: Point3d, allowedDrawingTypes: DrawingType[]) {
+    if (!viewport)
       return false;
-    for (const drawing of DrawingDataCache.getInstance().getSheetDrawingDataForViewport(ev.viewport)) {
+    for (const drawing of DrawingDataCache.getInstance().getSheetDrawingDataForViewport(viewport)) {
       if (allowedDrawingTypes.includes(drawing.type)) {
-        if (SheetMeasurementsHelper.checkIfInDrawing(ev.point, drawing.origin, drawing.extents)) {
+        if (SheetMeasurementsHelper.checkIfInDrawing(point, drawing.origin, drawing.extents)) {
           return true;
         }
       }
     }
     return false;
   }
+
+  function getNameFromDrawingType(type: SheetMeasurementsHelper.DrawingType): string  {
+    if (type === SheetMeasurementsHelper.DrawingType.CrossSection) {
+      return IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MeasurementNames.CrossSection");
+    } else if (type === SheetMeasurementsHelper.DrawingType.Plan) {
+      return IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MeasurementNames.Plan");
+    } else if (type === SheetMeasurementsHelper.DrawingType.Profile) {
+      return IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MeasurementNames.Profile");
+    } else {
+      return "";
+    }
+  }
+
+  export async function getSheetErrorToolTipText(hit: HitDetail, allowedDrawingTypesList: SheetMeasurementsHelper.DrawingType[], defaultToolTip:(hit: HitDetail) => Promise<HTMLElement | string>): Promise<string | HTMLElement> {
+    if (SheetMeasurementsHelper.checkIfAllowedDrawingType(hit.viewport, hit.hitPoint, allowedDrawingTypesList)) {
+      return defaultToolTip(hit);
+    }
+
+    if (allowedDrawingTypesList.length < 1) {
+      return "";
```

Comment:
Since a zero allowed drawing type list won't be allowed, it feels like then we need an error message for this case. I suppose realistically, it's a case that shouldn't happen to the end user (if no allowed types, then the sheet measurement would be disabled in the tool?). So this may actually be more of an error message to inform the developer that they did something wrong.

"Hey Jerk, you didn't add allow any drawing types" 😃

Reply:
So we're thinking more like throwing an error than a message here?